emmet-core/emmet/core/vasp/calculation.py

```diff
@@ -341,12 +341,21 @@ class IonicStep(BaseModel):  # type: ignore
     electronic_steps: Optional[List[ElectronicStep]] = Field(
         None, description="The electronic convergence steps."
     )
+    num_electronic_steps: Optional[int] = Field(
+        None, description="The number of electronic steps needed to reach convergence."
+    )
     structure: Optional[Structure] = Field(
         None, description="The structure at this step."
     )
 
     model_config = ConfigDict(extra="allow")
 
+    @model_validator(mode="after")
+    def set_elec_step_count(self):
+        if self.electronic_steps is not None and self.num_electronic_steps is None:
+            self.num_electronic_steps = len(self.electronic_steps)
+        return self
```

Comment:
maybe handle the case where `num_electronic_steps` is set but doesn't match `len(electronic_steps)`. probably `len(electronic_steps)` should take precedence or an error be raised if there's such a thing as a strict mode

Reply:
could also validate that `num_electronic_steps` is non-negative